File: 프로그래머스/멀쩡한 사각형.py
# 최대 공약수 구하기
def gcd(a, b):
    if b > a:
        a, b = b, a
    while b > 0:
        a, b = b, a % b
    return a

def solution(w,h):
    return (w * h) - (w + h - gcd(w,h)) 

File: 프로그래머스/멀쩡한 사각형.java
class Solution {
    private int gcd(int a, int b) {
        while (b != 0) {
            int r = a % b;
            a = b;
            b = r;
        }
        return a;
    }
    public long solution(int w, int h) {
        long count = gcd(w, h);
        long all = (long) w * h;
        return all - (w/count + h/count - 1) * count;
    }
}